eventbus/kafka/eventbus.go

```diff
@@ -0,0 +1,471 @@
+// Copyright (c) 2021 - The Event Horizon authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/segmentio/kafka-go"
+
+	eh "github.com/looplab/eventhorizon"
+	"github.com/looplab/eventhorizon/codec/json"
+)
+
+// EventBus is a local event bus that delegates handling of published events
+// to all matching registered handlers, in order of registration.
+type EventBus struct {
+	// TODO: Support multiple brokers.
+	addresses       []string
+	appID           string
+	autoCreateTopic bool
+	topic           string
+	topicPartitions int
+	startOffset     int64
+	client          *kafka.Client
+	writer          *kafka.Writer
+	registered      map[eh.EventHandlerType]struct{}
+	registeredMu    sync.RWMutex
+	errCh           chan error
+	cctx            context.Context
+	cancel          context.CancelFunc
+	wg              sync.WaitGroup
+	codec           eh.EventCodec
+}
+
+// NewEventBus creates an EventBus, with optional settings.
+func NewEventBus(addressList, appID string, options ...Option) (*EventBus, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	addrSplit := strings.Split(addressList, ",")
+
+	b := &EventBus{
+		addresses:       addrSplit,
+		appID:           appID,
+		autoCreateTopic: true,
```

Comment:
Do we want this to default to false?

Reply:
No, I'm leaving all the defaults the same as the original implementation so that this can serve as a drop-in replacement.